test/multiverse/suites/active_record_pg/external_request_from_within_ar_block_test.rb

```diff
@@ -0,0 +1,55 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require 'net/http'
+
+class ExternalRequestFromWithinARBlockTest < Minitest::Test
+  # Use the agent's segment callback system to register a callback for the
+  # ExternalRequestSegment class. Every time that class is initialized, the
+  # callback will be called and it will check to see if the external request
+  # segment has been created from within an ActiveRecord transaction block.
+  # If that check succeeds, generate an error and have the agent notice it.
+  def test_callback_to_notice_error_if_an_external_request_is_made_within_an_ar_block
+    callback = proc do
+      return unless caller.any? { |line| line.match?(%r{active_record/transactions.rb}) }
```

Comment:
As a feature request author, just would like to understand why this was chosen over a cleaner `ActiveRecord::Base.connection.transaction_open?`? Legacy compatibility?

Reply:
Hi @toncid, the current (version 9.x) agent no longer support Rails 3, so we can definitely make use of `transaction_open`. Updated with [8b01b16](https://github.com/newrelic/newrelic-ruby-agent/pull/2178/commits/8b01b16ba9ac17bdd16aa085a8ff904b74313cd0).